core/embed/rust/src/ui/display/font.rs

```diff
@@ -144,6 +144,36 @@ impl Font {
         display::text_width(text, self.into())
     }
 
+    /// Supports UTF8 characters
+    fn get_glyph_from_char(self, c: char) -> Option<Glyph> {
+        let mut buffer = [0; 4];
+        let bytes = c.encode_utf8(&mut buffer);
+        for byte in bytes.bytes() {
+            if let Some(glyph) = self.get_glyph(byte) {
+                return Some(glyph);
+            }
+        }
+        None
+    }
+
+    /// Supports UTF8 characters
+    fn get_first_glyph_from_text(self, text: &str) -> Option<Glyph> {
+        if let Some(c) = text.chars().next() {
+            self.get_glyph_from_char(c)
+        } else {
+            None
+        }
+    }
+
+    /// Supports UTF8 characters
+    fn get_last_glyph_from_text(self, text: &str) -> Option<Glyph> {
+        if let Some(c) = text.chars().next_back() {
+            self.get_glyph_from_char(c)
+        } else {
+            None
+        }
+    }
```

Comment:
this is completely weird. why not modify `get_glyph` to accept a `char` directly? (and then as @prusnak suggests, modify `get_char_glyph` to accept uint32)